app/src/main/java/com/example/bucketbuds/ProfileFragment.java

```diff
@@ -1,65 +1,295 @@
+        }
+        tvFirst.setText(user.getFirstName());
+        tvLast.setText(user.getLastName());
+        tvUsername.setText("@"+user.getUsername());
+        tvFriendCount.setText(String.valueOf(user.getFriendCount()));
+        tvBucketCount.setText(String.valueOf(user.getBucketCount()));
+
+        String bio = user.getBio();
+        if (bio == null) {
+            tvBio.setText(0);
+        } else {
+            tvBio.setText(bio);
+        }
+
+        btnEditProfile.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showEditProfileDialog();
+            }
+        });
+
+        ivProfileImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onLaunchCamera(v);
+            }
+        });
+
+        return view;
+    }
+
+    public void showEditProfileDialog(){
+        View messageView = LayoutInflater.from(getContext()).
+                inflate(R.layout.dialog_edit_profile, null);
+        // Create alert dialog builder
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
+        alertDialogBuilder.setView(messageView);
+        // Create alert dialog
+        final AlertDialog alertDialog = alertDialogBuilder.create();
+        EditText etUsername = messageView.findViewById(R.id.etUsernameDE);
+        EditText etBio = messageView.findViewById(R.id.etBioDE);
+        EditText etFirstName = messageView.findViewById(R.id.etFirstNameDE);
+        EditText etLastName = messageView.findViewById(R.id.etLastNameDE);
+        EditText etEmail = messageView.findViewById(R.id.etEmailDE);
+        Button btnSave = messageView.findViewById(R.id.btnSaveDE);
+        Button btnCancel = messageView.findViewById(R.id.btnCancelDE);
+
+        etBio.setText(user.getBio(), TextView.BufferType.EDITABLE);
+        etUsername.setText(user.getUsername(), TextView.BufferType.EDITABLE);
+        etFirstName.setText(user.getFirstName(), TextView.BufferType.EDITABLE);
+        etLastName.setText(user.getLastName(), TextView.BufferType.EDITABLE);
+        etEmail.setText(user.getEmail(), TextView.BufferType.EDITABLE);
+
+        // Configure dialog buttons
+        btnSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                String username = etUsername.getText().toString();
+                String bio = etBio.getText().toString();
+                String first = etFirstName.getText().toString();
+                String last = etLastName.getText().toString();
+                String email = etEmail.getText().toString();
+
+                if (username.isEmpty() || first.isEmpty() || last.isEmpty() || email.isEmpty()) {
+                    Toast.makeText(getContext(), "Name, username, and email are required", Toast.LENGTH_SHORT);
+                } else {
```

Comment:
Another thing that helps with nesting is early-exiting.
Instead of doing
```
if(...) {
  doSomething();
} else {
  doElse();
}
```
you can sometimes do
```
if(...) {
  doSomething();
  return;
}
doElse();
```